File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/base/constraints/bottomsheet/BottomSheetDialogConstraints.java
package com.github.RooneyAndShadows.lightbulb.dialogs.base.constraints.bottomsheet;

public class BottomSheetDialogConstraints {
    private final int minHeight;
    private final int maxHeight;

    public BottomSheetDialogConstraints(int minHeight, int maxHeight) {
        this.minHeight = minHeight;
        this.maxHeight = maxHeight;
    }

    public int resolveHeight(int desiredHeight) {
        if (desiredHeight > getMaxHeight())
            desiredHeight = getMaxHeight();
        if (desiredHeight < getMinHeight())
            desiredHeight = getMinHeight();
        return (desiredHeight);
    }

    public int getMinHeight() {
        return Math.min(minHeight, maxHeight);
    }

    public int getMaxHeight() {
        return Math.max(maxHeight, minHeight);
    }
}

File: lightbulb-dialogs/src/main/java/com/github/rooneyandshadowss/lightbulb/dialogs/base/constraints/bottomsheet/BottomSheetDialogConstraints.kt
package com.github.rooneyandshadowss.lightbulb.dialogs.base.constraints.bottomsheet

import kotlin.math.max
import kotlin.math.min

class BottomSheetDialogConstraints(private val minHeight: Int, private val maxHeight: Int) {
    fun resolveHeight(desiredHeight: Int): Int {
        var desired = desiredHeight
        if (desired > getMaxHeight())
            desired = getMaxHeight()
        if (desired < getMinHeight())
            desired = getMinHeight()
        return desired
    }

    fun getMinHeight(): Int {
        return min(minHeight, maxHeight)
    }

    fun getMaxHeight(): Int {
        return max(maxHeight, minHeight)
    }
}